File: Lab5/VkontakteRecycleVIew/app/src/main/java/com/example/vkrv/LockableViewPager.java
package com.example.vkrv;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.viewpager.widget.ViewPager;

public class LockableViewPager extends ViewPager {
    private boolean swipable;

    public LockableViewPager(Context context) {
        super(context);
    }

    public LockableViewPager(Context context, AttributeSet attrs) {
        super(context, attrs);
        this.swipable = true;
    }

    public boolean isSwipable() {
        return swipable;
    }

    public void setSwipable(boolean swipable) {
        this.swipable = swipable;
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        if(this.swipable)
        {
            return super.onInterceptTouchEvent(ev);
        }
        return false;
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        if(this.swipable){
            return super.onTouchEvent(ev);
        }
        return false;
    }
}

File: LabsInKotlin/Lab5/app/src/main/java/com/example/lab5/LockableViewPager.kt
package com.example.lab5

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.viewpager.widget.ViewPager


class LockableViewPager : ViewPager {
    private var isSwipable = false

    constructor(context: Context?) : super(context!!) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(
        context!!,
        attrs
    ) {
        isSwipable = true
    }

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        return if (isSwipable) {
            super.onInterceptTouchEvent(ev)
        } else false
    }

    override fun onTouchEvent(ev: MotionEvent): Boolean {
        return if (isSwipable) {
            super.onTouchEvent(ev)
        } else false
    }
    fun setSwipable(swipe: Boolean) {
        isSwipable = swipe
    }
}
